include/Dialects/LinalgTransform/FunctionalTransforms.h

```diff
@@ -0,0 +1,77 @@
+//===-- FunctionalTransforms.h - Functional Linalg Transforms -------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef IREE_LLVM_SANDBOX_DIALECTS_LINALGTRANSFORM_FUNCTIONALTRANSFORMS_H
+#define IREE_LLVM_SANDBOX_DIALECTS_LINALGTRANSFORM_FUNCTIONALTRANSFORMS_H
+
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/PatternMatch.h"
+
+namespace mlir {
+namespace linalg {
+
+class FunctionalPaddingPattern {
```

Comment:
I did some refactoring upstream and you should soon be able to just:
```
FailureOr<LinalgOp> applyPaddingPattern(
    LinalgOp op, PatternRewriter&rewriter, const LinalgPaddingOptions &options) {
  LinalgPaddingPattern pattern(op->getContext(), options);
  return pattern.returningMatchAndRewrite(op, rewriter);
}
```
everywhere here and below.

Reply:
The default empty filter will be passed and we wouldn't need to duplicate all those lines.
Do you have further extensions planned that wouldn't work with this?